src/sagemaker/workflow/utilities.py

```diff
@@ -151,3 +158,41 @@ def validate_step_args_input(
         raise TypeError(error_message)
     if step_args.caller_name not in expected_caller:
         raise ValueError(error_message)
+
+
+def override_pipeline_parameter_var(func):
+    """A temporary Decorator
+
+    This is a temporary decorator to override pipeline Parameter objects with their default value
+    and display warning information to instruct users to update their code.
+
+    This decorator can help to give a grace period for users to update their code when
+    we make changes to explicitly prevent passing any pipeline variables to a function.
+
+    We should remove this decorator after the grace period.
+    """
+    warning_info_suffix = (
+        "The default_value of this Parameter object will be used to override it. "
+        "Please remove this pipeline variable and use python primitives instead."
+    )
+
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        args = list(args)
+        for idx, arg in enumerate(args):
+            if isinstance(arg, Parameter):
+                warn_msg = (
+                    "Pipeline variable (type: %s) is not allowed here. " + warning_info_suffix
```

Comment:
can we print out the actual name of the arg that we are preventing being a pipeline param here? like `instance_type`

Reply:
Updated in the new commit